JustSaying/AwsTools/QueueCreation/RegionResourceCache.cs

```diff
@@ -1,33 +1,30 @@
-using System.Collections.Generic;
-
-namespace JustSaying.AwsTools.QueueCreation
-{
-    public class RegionResourceCache<T> : Dictionary<string, Dictionary<string, T>>, IRegionResourceCache<T>
-    {
-        public T TryGetFromCache(string region, string key)
-        {
-            if (! ContainsKey(region))
-            {
-                return default(T);
-            }
-
-            var regionDict = this[region];
-            if (! regionDict.ContainsKey(key))
-            {
-                return default(T);
-            }
-
-            return regionDict[key];
-        }
-
-        public void AddToCache(string region, string key, T value)
-        {
-            if (!ContainsKey(region))
-            {
-                this[region] = new Dictionary<string, T>();
-            }
-            var regionDict = this[region];
-            regionDict[key] = value;
-        }
-    }
-}
\ No newline at end of file
+using System;
+using System.Collections.Concurrent;
+
+namespace JustSaying.AwsTools.QueueCreation
+{
+    public sealed class RegionResourceCache<T>
+    {
+        private readonly ConcurrentDictionary<string, ConcurrentDictionary<string, T>> _regionsData
+            = new ConcurrentDictionary<string, ConcurrentDictionary<string, T>>(StringComparer.InvariantCultureIgnoreCase);

```

Comment:
`OrdinalIgnoreCase` should be used, I believe, if memory serves this one will raise a different analyzer warning as I'm 99% sure the equivalent `StringComparison` does if you don't use one of the two ordinal ones.

Reply:
Explanation: https://stackoverflow.com/a/72766/1064169